### PR TITLE
Set up the multi-active-gw branch

### PR DESCRIPTION
**Pull request by skitt**
This sets up CODEOWNERS so that the multi-active-gateway developers
can merge PRs to this branch independently.

It also updates the branch check to enforce use of the multi-active-gw
branch.

See <https://github.com/submariner-io/submariner/issues/1731> for
details.

Depends on https://github.com/submariner-io/shipyard/pull/761
Depends on https://github.com/submariner-io/shipyard/pull/759
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by tpantelis — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+/.github/workflows/ @anfredette @astoycos @Billy99 @mkolesnik @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+/package/ @anfredette @astoycos @Billy99 @mkolesnik @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+/scripts/ @anfredette @astoycos @Billy99 @mkolesnik @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+branch # @anfredette @astoycos @Billy99 @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+codeowners # @anfredette @astoycos @Billy99 @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+Core # @anfredette @astoycos @Billy99 @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+Dockerfile.dapper @anfredette @astoycos @Billy99 @mkolesnik @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+Feature # @anfredette @astoycos @Billy99 @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+Makefile* @anfredette @astoycos @Billy99 @mkolesnik @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
+Submariner # @anfredette @astoycos @Billy99 @nerdalert @Oats87 @skitt @sridhargaddam @tpantelis
```
Comment (sridhargaddam) [inline CODEOWNERS]: Looks like line 9, 11 and 13 are added because of comments in CODEOWNERS.in (Line 1 and 9). Can we fix it?

Comment (skitt) [reply CODEOWNERS]: Indeed, thanks! https://github.com/submariner-io/shipyard/pull/761 will fix this.

**Review by sridhargaddam — APPROVED** [review]
Thanks @skitt

**Review by dfarrell07 — APPROVED** [review]

**Comment by dfarrell07** [comment]
This seems ready to merge?

**Pull request merged**